### PR TITLE
Update/Add git upstream to an existing project

### PR DESCRIPTION
**Pull request by checkelmann**
Depends on github.com/keptn/go-utils/pull/169
Resolves Issue https://github.com/keptn/keptn/issues/1517

### CODE REVIEW

**Comment by checkelmann** [comment]
@johannes-b @bacherfl Could you take a look into it if I should change something?

File: cli/cmd/update_project.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+var upProjectCmd = &cobra.Command{
+	Use:   "project PROJECTNAME --git-user=GIT_USER --git-token=GIT_TOKEN --git-remote-url=GIT_REMOTE_URL",
+	Short: "Updates an existing keptn project",
+	Long: `Updates an existing keptn project with the provided name. 
+Updating a shipyard file is not possible.
+
+By executing the *update project* command, keptn will add the provided upstream repository to the existing internal Git repository that is used to maintain all project-related resources. 
+To upstream this internal Git repository to a remote repository, the Git user (*--git-user*), an access token (*--git-token*), and the remote URL (*--git-remote-url*) are required.
+
+For more information about updateing projects or upstream repositories visit https://keptn.sh/docs/develop/manage/project/ .
```
Comment (johannes-b) [inline cli/cmd/update_project.go:35]: "updateing" --> updating

File: cli/cmd/update_project.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+			errorMsg += "Keptn relies on the following conventions: "
+			errorMsg += "start with a lower case letter, then lower case letters, numbers, and hyphens are allowed.\n"
+			errorMsg += "Please update project name and try again."
+			return errors.New(errorMsg)
+		}
+		return nil
+	},
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+
+		if *updateProjectParams.GitUser == "" && *updateProjectParams.GitToken == "" && *updateProjectParams.RemoteURL == "" {
```
Comment (johannes-b) [inline cli/cmd/update_project.go:61]: Is there a reason for this check? I think that a project update without those parameters does not make sense.

**Comment by johannes-b** [comment]
Hi, I reviewed the code and it looks good from my perspective (please just address the two commands). However, let's wait for @bacherfl's feedback as he knows best the details about locking a project when executing a git command.

**Pull request merged**